test/core/util/erlang_log_translator_test.exs

```diff
@@ -0,0 +1,59 @@
+# Copyright(c) 2015-2021 ACCESS CO., LTD. All rights reserved.
+
+use Croma
+
+defmodule AntikytheraCore.ErlangLogTranslatorTest do
+  use Croma.TestCase
+  alias AntikytheraCore.ExecutorPool.ActionRunner
+
+  setup do
+    :meck.new(ErlangLogTranslator, [:passthrough])
+    on_exit(&:meck.unload/0)
+  end
+
+  test "should neglect SASL progress report from supervisor" do
+    # Use ActionRunner as just a GenServer, not as a PoolSup worker
+    children = [{ActionRunner, [nil]}]
+    {:ok, _pid} = Supervisor.start_link(children, strategy: :one_for_one, name: :test_supervisor)
+
+    # ensure logging
+    Process.sleep(100)
+
+    [{_pid, {ErlangLogTranslator, :translate, args}, return}] = :meck.history(ErlangLogTranslator)
+
+    supervisor =
+      if System.otp_release() >= "21" do
+        [_min_level, :info, :report, {{:supervisor, :progress}, data}] = args
```

Comment:
Since we assume that this format is coming from Erlang, it would be better to add `assert` to this pattern match although we can realize this assertion failure because Elixir raises `MatchError`.

My simple test case returns the following without `assert`
```
  1) test ExUnit pattern match (ElixirTestTest)
     test/elixir_test_test.exs:9
     ** (MatchError) no match of right hand side value: %{hoge: "fuga"}
     code: %{hoge1: str} = inside_map
     stacktrace:
       test/elixir_test_test.exs:12: (test)
```

and returns the following with `assert`
```
  1) test ExUnit pattern match (ElixirTestTest)
     test/elixir_test_test.exs:9
     match (=) failed
     code:  assert %{hoge1: str} = inside_map
     right: %{hoge: "fuga"}
     stacktrace:
       test/elixir_test_test.exs:12: (test)
```

Reply:
Thank you.
By adding `assert`, intention of the test becomes clear!

fix at https://github.com/access-company/antikythera/pull/173/commits/612001117464f3c3c32cc157fe191b02a59b8a25